freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -92,6 +92,10 @@ def handle(self, event):
             self.db_event.transition(EventState.SKIPPED, msg)
             return []
 
+        if isinstance(self.event, ManualBundleRebuildEvent) and self.event.manual_image_remap:
+            log.info("Performing manual NVR remapping")
+            self._manual_bundle_remap(bundles_to_rebuild)  #inplace substitutions  to bundles_to_rebuild
```

Comment:
We are not overriding the bundles to be rebuilt, we want to override the related images in bundle while rebuilding the bundles.

When a rebuilds advisory (filed by Botas) is shipped, it has a list of images which were built by Freshmker, for each image, it has corresponding original image, so we can query Errata and Freshmaker DB to get some data like this:
```
original_nvr1 -> rebuilt_nvr1
original_nvr2 -> rebuilt_nvr2
original_nvr3 -> rebuilt_nvr3
```
the `rebuilt_nvr*` are images in the advisory.

Then in this handler, Freshmaker will try to find the bundles that are referencing the following images:
```
original_nvr1
original_nvr2
original_nvr3
```
If any bundle is referencing any of the listed original images, then Freshmaker will rebuild that bundle by:
1. replace `original_nvr1` with `rebuilt_nvr1`
2. replace `original_nvr2` with `rebuilt_nvr2`
3. replace `original_nvr3` with `rebuilt_nvr3`

Our purpose here is to support overriding the mapping, for example, if we have something like:
```
'bundle_related_image_overrides': {'original_nvr3': 'some_nvr3', 'original_nvr4': 'some_nvr4'}
```
then we should be able to override the mapping to:
```
original_nvr1 -> rebuilt_nvr1
original_nvr2 -> rebuilt_nvr2
original_nvr3 -> some_nvr3
original_nvr4 -> some_nvr4
```
we can override the mappings after https://github.com/redhat-exd-rebuilds/freshmaker/blob/9788db751ee6c94c6b59c01464622c53c4bdc494/freshmaker/handlers/botas/botas_shipped_advisory.py#L144

But to prevent unexpected rebuilds, I'd suggest only allow the overrides when bundle images (`container_images` in request) are specified explicitly in the request, we can add the check in https://github.com/redhat-exd-rebuilds/freshmaker/blob/9788db751ee6c94c6b59c01464622c53c4bdc494/freshmaker/views.py#LL361C7-L361C7

Reply:
I see! Thanks for the detailed explanation!